pyflakes/test/test_type_annotations.py

```diff
@@ -695,3 +695,24 @@ class C(Protocol):
                 def f():  # type: () -> int
                     pass
         """)
+
+    @skipIf(version_info < (3, 6), 'new in Python 3.6')
+    def test_typeddict_allows_any_names(self):
+        self.flakes("""
+            from typing import TypedDict, List, NamedTuple
+
+            List[TypedDict("x", {})]
+            List[NamedTuple("a", a=int)]
+            List[NamedTuple("a", [("a", int)])]
+
+            class X(TypedDict):
+                y: TypedDict("z", {"zz":int})
```

Comment:
this is the only syntax which is python3.6-specific -- the other tests should be tested in all the other python versions as well